test-scripts/resources-tests.sh

```diff
@@ -0,0 +1,551 @@
+
+    info "Command is: $cmd"
+
+    $cmd
+    rc=$?
+
+    rm $input_file
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to update resources"
+        exit 1
+    fi
+
+    # check that property resource has been updated
+    cmd="$ORIGINAL_DIR/bin/${binary} properties get --namespace ecosystemtest \
+    --name $prop_name \
+    --bootstrap $bootstrap \
+    --log -"
+
+    info "Command is: $cmd"
+
+    output_file="$ORIGINAL_DIR/temp/resources-update-output.txt"
+    $cmd | tee $output_file
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to get property with name used, get command failed."
+        exit 1
+    fi
+
+    # Check that the previous properties set updated the property value
+    cat $output_file | grep "$prop_name\s+$prop_value" -q -E
+
+    rc=$?
+    # We expect a return code of 0 because this is a properly formed properties get command.
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to update property, property not found in namespace."
+        exit 1
+    fi
+
+    rm $output_file
+
+    success "Resources update seems to be successful."
+}
+
+#--------------------------------------------------------------------------
+function properties_resources_apply {
+    h2 "Performing resources apply, expecting ..."
+
+    used -o pipefail # Fail everything if anything in the pipeline fails. Else we are just checking the 'tee' return code.
+
+    prop_name_to_create="properties.test.create"
+    prop_value_to_create="value-created"
+    #this property needs to already exist
+    prop_name_to_update="properties.test.1"
+    prop_value_to_update="updated-value-2"
+
+    input_file="$ORIGINAL_DIR/temp/resources-apply-input.yaml"
+
+    echo """apiVersion: galasa-dev/v1alpha1
+kind: GalasaProperty
+metadata:
+  name: $prop_name_to_update
+  namespace: ecosystemtest
+data:
+  value: $prop_value_to_update
+---
+apiVersion: galasa-dev/v1alpha1
+kind: GalasaProperty
+metadata:
+  name: $prop_name_to_create
+  namespace: ecosystemtest
+data:
+  value: $prop_value_to_create
+""" > $input_file
+
+    cmd="$ORIGINAL_DIR/bin/${binary} resources apply \
+    --bootstrap $bootstrap \
+    -f $input_file \
+    --log -"
+
+    info "Command is: $cmd"
+
+    $cmd
+    rc=$?
+
+    rm $input_file
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to apply resources"
+        exit 1
+    fi
+
+    # check that property resource has been applied
+    cmd="$ORIGINAL_DIR/bin/${binary} properties get --namespace ecosystemtest \
+    --infix properties.test \
+    --bootstrap $bootstrap \
+    --log -"
+
+    info "Command is: $cmd"
+
+    output_file="$ORIGINAL_DIR/temp/resources-apply-output.txt"
+    $cmd | tee $output_file
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to get property with name used, get command failed."
+        exit 1
+    fi
+
+    # Check that the previous properties set updated the property value
+    cat $output_file | grep "$prop_name_to_update\s+$prop_value_to_update" -q -E
```

Comment:
why do this twice ?

Reply:
This is to check that both properties in the file have been applied (one created, and another updated) and have the expected values